lighthouse-core/fraggle-rock/config/config.js

```diff
@@ -241,20 +239,21 @@ function resolveNavigationsToDefns(navigations, artifactDefns, settings) {
 
 /**
  * @param {LH.Config.Json|undefined} configJSON
- * @param {ConfigContext} context
+ * @param {LH.Flags} flags
+ * @param {LH.Gatherer.GatherMode} gatherMode
  * @return {{config: LH.Config.FRConfig, warnings: string[]}}
  */
-function initializeConfig(configJSON, context) {
+function initializeConfig(configJSON, flags, gatherMode) {
```

Comment:
wdyt about

```suggestion
function initializeConfig(configJSON, gatherMode, flags = {}) {
```

it seems like we ALWAYS are providing gathermode but a million tests have an empty obj for flags.

in order of importance, i think the mode wins, too.

Reply:
I'm fine with this, but I would argue that `gatherMode` belongs before `configJSON` too because `configJSON` can be undefined as well.